### PR TITLE
Idex 3006

### PR DESCRIPTION
**Pull request by vparfonov**

### CODE REVIEW

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectTypes.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import org.eclipse.che.api.project.server.type.ProjectType;
+import org.eclipse.che.api.project.shared.dto.SourceEstimation;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+/**
```
Comment (skabashnyuk) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectTypes.java]: Javadoc

File: platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/FactoryService.java [hunk 2]
```diff
@@ -594,40 +594,36 @@ public Response getFactoryJson(@ApiParam(value = "Workspace ID", required = true
...
+                Map<String, List<String>> moduleAttributes = new HashMap<>();
+                for (Map.Entry<String, AttributeValue> entry : moduleConfig.getAttributes().entrySet()) {
+                    projectAttributes.put(entry.getKey(), entry.getValue().getList());
+                }
+
                 newProject.getModules().add(DtoFactory.newDto(ProjectModule.class).withType(moduleConfig.getTypeId())
                                                       .withPath(moduleRelativePath)
                                                       .withRecipe(moduleConfig.getRecipe())
-                                                      .withAttributes(moduleJson.getAttributes())
+                                                      .withAttributes(moduleAttributes)
```
Comment (skabashnyuk) [inline platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/FactoryService.java]: projectDescription.getAttributes().entrySet().stream().collect(Collectors.toMap( Map.Entry :: getKey, v -> v.getValue().getList())); 
Wdyt?

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/Constants.java [hunk 3]
```diff
@@ -13,24 +13,20 @@
...
+    public static final String LINK_REL_GET_CONTENT    = "get content";
+    public static final String LINK_REL_UPDATE_CONTENT = "update content";
 
     public static final String LINK_REL_PROJECT_TYPES = "project types";
 
-    public static final String CODENVY_DIR                        = ".codenvy";
-    public static final String CODENVY_PROJECT_FILE               = "project.json";
-    public static final String CODENVY_PROJECT_FILE_RELATIVE_PATH = CODENVY_DIR + "/" + CODENVY_PROJECT_FILE;
-//    public static final String CODENVY_RUNNER_ENVIRONMENTS_DIR    = CODENVY_DIR + "/runners/environments";
+    public static final String CODENVY_DIR = ".codenvy";
```
Comment (skabashnyuk) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/Constants.java]: You sure we need it? If yes may be we should call it .che?

Comment (skabashnyuk) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/Constants.java]: BTW 
public static final String CHE_DIR

Comment (vparfonov) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/Constants.java]: good question 4.0 good time to change it

File: platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/FactoryService.java (hunk 2 above)
Comment (azatsarynnyy) [inline platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/FactoryService.java]: Much more better. Thx
Updated

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/Constants.java (hunk 3 above)
Comment (vparfonov) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/Constants.java]: will be fixed in other branch

**Pull request merged**